test/server/spec/lib/args.spec.js

```diff
@@ -130,35 +132,82 @@ describe("lib/args", function () {
         queue: null,
         buffer: true,
         tries: 1,
-        setup: null
+        setup: null,
+        bail: true
       });
 
       expect(_flags(args.concurrent(argv.concat(["--no-buffer"])))).to.deep.equal({
         queue: null,
         buffer: false,
         tries: 1,
-        setup: null
+        setup: null,
+        bail: true
       });
 
       expect(_flags(args.concurrent(argv.concat(["--buffer=false"])))).to.deep.equal({
         queue: null,
         buffer: false,
         tries: 1,
-        setup: null
+        setup: null,
+        bail: true
       });
 
       expect(_flags(args.concurrent(argv.concat(["--buffer=true"])))).to.deep.equal({
         queue: null,
         buffer: true,
         tries: 1,
-        setup: null
+        setup: null,
+        bail: true
       });
 
       expect(_flags(args.concurrent(argv.concat(["--buffer=1"])))).to.deep.equal({
         queue: null,
         buffer: true,
         tries: 1,
-        setup: null
+        setup: null,
+        bail: true
+      });
+    });
+
+    it("handles valid --bail", function () {
+      expect(_flags(args.concurrent(argv.concat(["--bail"])))).to.deep.equal({
+        queue: null,
+        buffer: false,
+        tries: 1,
+        setup: null,
+        bail: true
+      });
+
+      expect(_flags(args.concurrent(argv.concat(["--no-bail"])))).to.deep.equal({
+        queue: null,
+        buffer: false,
+        tries: 1,
+        setup: null,
+        bail: false
+      });
+
+      expect(_flags(args.concurrent(argv.concat(["--bail=false"])))).to.deep.equal({
+        queue: null,
+        buffer: false,
+        tries: 1,
+        setup: null,
+        bail: false
+      });
+
+      expect(_flags(args.concurrent(argv.concat(["--bail=true"])))).to.deep.equal({
+        queue: null,
+        buffer: false,
+        tries: 1,
+        setup: null,
+        bail: true
+      });
+
+      expect(_flags(args.concurrent(argv.concat(["--bail=1"])))).to.deep.equal({
+        queue: null,
+        buffer: false,
+        tries: 1,
+        setup: null,
+        bail: true
```

Comment:
This test is somewhat misleading as it implies `--bail=0` works, but it doesn't (`nopt` will turn 0 into true).